servicetalk-transport-api/src/main/java/io/servicetalk/transport/api/IoExecutor.java

```diff
@@ -15,15 +15,15 @@
  */
 package io.servicetalk.transport.api;
 
+import io.servicetalk.concurrent.api.Executor;
 import io.servicetalk.concurrent.api.ListenableAsyncCloseable;
 
-import java.util.concurrent.Executor;
 import java.util.function.BooleanSupplier;
 
 /**
  * {@link Executor} that handles IO.
  */
-public interface IoExecutor extends ListenableAsyncCloseable {
+public interface IoExecutor extends Executor, ListenableAsyncCloseable {
```

Comment:
1. `io.servicetalk.concurrent.api.Executor` already implements `io.servicetalk.concurrent.api.ListenableAsyncCloseable`, explicit declaration can be removed.
2. This PR deprecates `NettyIoExecutor#asExecutor()`, meaning we will anyway need a follow-up PR to remove this deprecation for 0.43. In this case, consider adding `default` implementations for `IoExecutor#execute(...)` and `IoExecutor#schedule(...)` that will throw `UnsupportedOperationException` with a comment `FIXME: 0.43 - remove default methods`. It will make this PR ready for cherry-picking.